mono/utils/mono-threads-api.h

```diff
@@ -22,35 +22,77 @@ MONO_BEGIN_DECLS
 This API is experimental. It will eventually be required to properly use the rest of the raw-omp embedding API.
 */
 
+typedef struct _MonoStackData {
+	gpointer stackpointer;
+	const char *function_name;
+} MonoStackData;
+
+// FIXME an ifdef to change __func__ to empty or further minimization.
+#define MONO_STACKDATA(x) MonoStackData x = { &x, __func__ }
+
+static inline const char*
+mono_stackdata_get_function_name (const MonoStackData *stackdata)
+{
+	return stackdata->function_name;
+}
+
+static inline gpointer
+mono_stackdata_get_stackpointer (const MonoStackData *stackdata)
+{
+	return stackdata->stackpointer;
+}
+
 MONO_API gpointer
 mono_threads_enter_gc_unsafe_region (gpointer* stackdata);
 
+gpointer
+mono_threads_enter_gc_unsafe_region_internal (MonoStackData *stackdata);
+
 MONO_API void
 mono_threads_exit_gc_unsafe_region (gpointer cookie, gpointer* stackdata);
 
+void
+mono_threads_exit_gc_unsafe_region_internal (gpointer cookie, MonoStackData *stackdata);
+
 MONO_API gpointer
 mono_threads_enter_gc_unsafe_region_unbalanced (gpointer* stackdata);
 
+gpointer
+mono_threads_enter_gc_unsafe_region_unbalanced_internal (MonoStackData *stackdata);
+
 MONO_API void
 mono_threads_exit_gc_unsafe_region_unbalanced (gpointer cookie, gpointer* stackdata);
 
+void
+mono_threads_exit_gc_unsafe_region_unbalanced_internal (gpointer cookie, MonoStackData *stackdata);
+
 MONO_API void
 mono_threads_assert_gc_unsafe_region (void);
 
-
-
 MONO_API gpointer
 mono_threads_enter_gc_safe_region (gpointer *stackdata);
 
+MONO_API gpointer // used by profiler
```

Comment:
`MONO_PROFILER_API` is what you want.

Reply:
It is defined in `mono-compiler.h`

Reply:
Yes I saw that but it didn't compile and didn't seem worth fighting. I'll try again.

Reply:
I tried MONO_PROFILER_API again. It failed again.
It is common to include mono-threads-api.h. without mono-compiler.h.
So I added include mono-compiler.h, that I was avoiding before.